src/main.cpp

```diff
@@ -151,188 +237,158 @@ void Frame::OnTagSelectionChange(wxCommandEvent &event)
+            auto sv = std::string_view{contents};
+            auto skip_whitespace = [&](std::string_view& sv) {
+                auto pos = sv.find_first_not_of(whitespace);
+                if (pos != std::string_view::npos) { sv.remove_prefix(pos); }
+            };
 
             // Skip whitespace at beginnning.
-            size_t whitespace_at_beginning = strspn(tags_string, whitespace.data());
-            tags_string += whitespace_at_beginning;
+            skip_whitespace(sv);
+            if (sv.empty()) { continue; } // String is entirely whitespace.
 
-            if (memcmp(tags_string, tag_marker.data(), tag_marker.length()) == 0) {
+            if (sv.starts_with(tag_marker)) {
                 // Tags are at beginning.
                 // Skip past the tag marker.
-                tags_string += tag_marker.length();
+                sv.remove_prefix(tag_marker.size());
 
                 // Start to parse tags white-space separated, up until a newline is met.
                 // Find end (first newline)
-                char* end_of_tags = strstr(tags_string, "\n");
-                // TODO: Handle the case of end of file, no newline.
-
-                // Add new entry to entry_data vector.
-                Entry new_entry;
-                new_entry.filepath = path.string();
-                entry_data.push_back(new_entry);
-                Entry& entry = entry_data.back();
-                entry.index = entry_data.size() - 1;
-
-                while (tags_string < end_of_tags) {
-                    // Skip past all whitespace.
-                    size_t skip_amount = strspn(tags_string, whitespace.data());
-                    if (skip_amount == 0) {
-                        break;
-                    }
-                    tags_string += skip_amount;
-                    char *tag_begin = tags_string;
+                auto end_of_tags = sv.find_first_of('\n');
+                if (end_of_tags != std::string::npos) {
+                    sv.remove_suffix(sv.size() - end_of_tags);
+                }
 
+                // Create a new entry.
+                Entry entry;
+                entry.filepath = path;
+                entry.index = entry_data.size();
+
+                // Parse the tags.
+                while (skip_whitespace(sv), !sv.empty()) {
                     // Skip to the next whitespace.
-                    size_t tag_length = strcspn(tags_string, whitespace.data());
-                    tags_string += tag_length;
-                    char *tag_end = tags_string;
+                    auto end_of_tag = sv.find_first_of(whitespace);
+                    if (end_of_tag == std::string::npos) { end_of_tag = sv.size(); }
 
                     // Add the tag found to the set of tags.
-                    Tag new_tag(std::string(tag_begin, tag_end));
-                    Tag* tag = add_tag(tag_data, new_tag);
+                    Tag& tag = add_tag(tag_data, sv.substr(0, end_of_tag));
```

Comment:
Unlike `std::string::substr()`, `std::string_view::substr()` *doesn't* copy anything; it just creates a new `std::string_view`.